src/mpi/events.rs

```diff
@@ -19,6 +21,8 @@ pub trait EventsSubscriber {
     fn receive_event(&mut self, time: u32, event: &Event);
 
     fn finish(&mut self) {}
+
+    fn as_any(&self) -> &dyn Any;
```

Comment:
Very nice! I didn't know this pattern. 

Since every `EventsSubscriber` has the same implementation it probably makes sense to provide a [default implementation](https://doc.rust-lang.org/book/ch10-02-traits.html#default-implementations) right here.

Reply:
I tried that out, but cargo complains: 
```
error[E0277]: the size for values of type `Self` cannot be known at compilation time
  --> src/simulation/messaging/events.rs:26:9
   |
26 |         self
   |         ^^^^ doesn't have a size known at compile-time
   |
   = note: required for the cast from `Self` to the object type `dyn Any`
```

Reply:
btw here is the source of that pattern: https://stackoverflow.com/questions/33687447/how-to-get-a-reference-to-a-concrete-type-from-a-trait-object

Reply:
I see. Started to read about `any`, but I don't want to learn that right now. I can live with the [comment's](https://stackoverflow.com/a/33687996/9945451) explanation. One of the comments states that you can do this [directly on boxes](https://doc.rust-lang.org/std/any/trait.Any.html#impl-Box%3Cdyn%20Any%20+%20Send%20+%20%27static%2C%20A%3E) without `as_any` now. I haven't looked at it thorowly though. 

You can rather go ahead and finish this PR and we'll worry later once this annoys us too much.